2022/opmodes/JavaTeleop.java

```diff
@@ -0,0 +1,169 @@
+import team25core.StandardFourMotorRobot;
+import team25core.TankMechanumControlScheme;
+import team25core.TankMechanumControlSchemeReverse;
+import team25core.TeleopDriveTask;
+
+@TeleOp(name = "JavaTeleop")
+//@Disabled
+public class JavaTeleop extends StandardFourMotorRobot {
+
+
+    private Telemetry.Item buttonTlm;
+
+    private TeleopDriveTask drivetask;
+    private DcMotor carouselMech;
+    private DcMotor liftMotor;
+    private DcMotor intakeMotor;
+
+    private MechanumGearedDrivetrain drivetrain;
+
+    private static final int TICKS_PER_INCH = 79;
+
+    @Override
+    public void handleEvent(RobotEvent e) {
+    }
+
+    @Override
+    public void init() {
+
+        super.init();
+
+        //mechanisms
+        carouselMech = hardwareMap.get(DcMotor.class, "carouselMech");
+        liftMotor = hardwareMap.get(DcMotor.class,"liftMotor");
+        intakeMotor = hardwareMap.get(DcMotor.class, "intakeMotor");
+
+        // using encoders to record ticks
+
+        backLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        backRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        //telemetry
+        buttonTlm = telemetry.addData("buttonState", "unknown");
+
+        TankMechanumControlSchemeReverse scheme = new TankMechanumControlSchemeReverse(gamepad1);
+        drivetrain = new MechanumGearedDrivetrain(motorMap);
+        drivetrain.setCanonicalMotorDirection();
+        // Note we are swapping the rights and lefts in the arguments below
+        // since the gamesticks were switched for some reason and we need to do
+        // more investigation
+        drivetask = new TeleopDriveTask(this, scheme, backLeft, backRight, frontLeft, frontRight);
+    }
+
+    @Override
+    public void start() {
+
+        //Gamepad 1
+        this.addTask(drivetask);
+
+
+        //Gamepad 2
+        this.addTask(new GamepadTask(this, GamepadTask.GamepadNumber.GAMEPAD_2) {
+            //@Override
+            public void handleEvent(RobotEvent e) {
+                GamepadEvent gamepadEvent = (GamepadEvent) e;
+
+                switch (gamepadEvent.kind) {
+                    case BUTTON_X_DOWN:
+                        //enable carouselMech
+                        carouselMech.setDirection(DcMotorSimple.Direction.FORWARD);
+                        carouselMech.setPower(0.2);
+                        break;
+                    case BUTTON_X_UP:
+                    case BUTTON_Y_UP:
+                        //disable carouselMech
+                        carouselMech.setPower(0);
+                        break;
+                    case BUTTON_Y_DOWN:
+                        //enable carouselMech
+                        carouselMech.setDirection(DcMotorSimple.Direction.REVERSE);
+                        carouselMech.setPower(0.2);
+                        break;
+                    case LEFT_STICK_UP:
```

Comment:
So this is a rather clever addition to the Gamepad task, but this really smells more like a deadman motor usage which would call for a DeadmanMotorTask instead.

And while the DeadmanMotorTask currently only uses buttons for control I think that would be the place to extend to incorporate left/right joystick.